src/clients/security.rb

```diff
@@ -180,83 +179,66 @@ def SecuritySummaryHandler(options)
     end
 
     # Set value of specific security option
-    # @return [Boolean] false
+    # @return [Boolean] successfully modified?
     def SecuritySetHandler(options)
       options = deep_copy(options)
-      if Builtins.haskey(options, "passwd") &&
-          Ops.get_string(options, "passwd", "") !=
-            Ops.get(Security.Settings, "PASSWD_ENCRYPTION", "")
-        Ops.set(
-          Security.Settings,
-          "PASSWD_ENCRYPTION",
-          Ops.get_string(options, "passwd", "des")
-        )
+
+      if options.key?("password") &&
+          Security.Settings["PASSWD_ENCRYPTION"] != options["password"]
+        Security.Settings["PASSWD_ENCRYPTION"] = options.fetch("password", "des")
         Security.modified = true
       end
-      if Builtins.haskey(options, "crack") &&
-          Ops.get_string(options, "crack", "") !=
-            Ops.get(Security.Settings, "PASSWD_USE_CRACKLIB", "")
-        Ops.set(
-          Security.Settings,
-          "PASSWD_USE_CRACKLIB",
-          Ops.get_string(options, "crack", "yes")
-        )
+
+      if options_has_key?("crack") &&
+          Security.Settings["PASSWD_USE_CRACKLIB"] != options["crack"]
+        Security.Settings["PASSWD_USE_CRACKLIB"] = options.fetch("crack", "yes")
         Security.modified = true
       end
-      if Builtins.haskey(options, "permissions") &&
-          !Builtins.issubstring(
-            Ops.get(Security.Settings, "PERMISSION_SECURITY", ""),
-            Ops.get_string(options, "permissions", "")
-          )
-        Ops.set(
-          Security.Settings,
-          "PERMISSION_SECURITY",
-          Ops.add(Ops.get_string(options, "permissions", ""), " local")
-        )
+
+      if options.key?("permissions") &&
+          Security.Settings["PERMISSION_SECURITY"] != options["permissions"]
+        Security.Settings["PERMISSION_SECURITY"] = options.fetch("permissions", "local")
         Security.modified = true
       end
 
-      if Builtins.haskey(options, "remember") &&
-          Ops.get(Security.Settings, "PASSWD_REMEMBER_HISTORY", "0") !=
-            Ops.get_string(options, "remember", "0")
-        to_remember = Builtins.tointeger(
-          Ops.get_string(options, "remember", "0")
-        )
-        if to_remember == nil || Ops.less_than(to_remember, 0) ||
-            Ops.greater_than(to_remember, 400)
-          # error message
+      if options.key?("remember") &&
+          Security.Settings["PASSWD_REMEMBER_HISTORY"] != options["remember"]
+        if options["remember"].to_i.between?(0, 400)
+          Security.Settings["PASSWD_REMEMBER_HISTORY"] = options["remember"].to_i
+          Security.modified = true
+        else
           Report.Error(
             _("The number of passwords to remember must be between 0 an 400.")
           )
-          return false
         end
-        Ops.set(
-          Security.Settings,
-          "PASSWD_REMEMBER_HISTORY",
-          Ops.get_string(options, "remember", "0")
-        )
```

Comment:
I changed it already to use strings. I think there is no problem, but consistence is good.